modules/tinymce/src/themes/silver/main/ts/ui/skin/Loader.ts

```diff
@@ -23,58 +39,97 @@ const loadRawCss = (editor: Editor, key: string, css: string, styleSheetLoader:
   return styleSheetLoader.loadRawCss(key, css);
 };
 
-const loadUiSkins = async (editor: Editor, skinUrl: string): Promise<void> => {
-  const skinResourceIdentifier = Options.getSkinUrlOption(editor).getOr('default');
-  const skinUiCss = 'ui/' + skinResourceIdentifier + '/skin.css';
-  const css = tinymce.Resource.get(skinUiCss);
-  if (Type.isString(css)) {
-    loadRawCss(editor, skinUiCss, css, editor.ui.styleSheetLoader);
-  } else {
+const skinIdentifierToResourceKey = (identifier: string, filename: string): string =>
+  'ui/' + identifier + '/' + filename;
+
+const getResourceValue = (resourceKey: string): Optional<string> =>
+  Optional.from(tinymce.Resource.get(resourceKey)).filter(Type.isString);
+
+const loadCSS = (
+  editor: Editor,
+  filenameBase: string,
+  onLoadRaw: (key: string, css: string) => Promise<void>,
+  onLoadStyleSheet: (styleSheetUrl: string) => Promise<void>,
```

Comment:
There was a time when we'd use a proper `ADT` for this instead of approximating it with optional some/none 🥲

Reply:
A typescript-based ADT is doable. I am not sure if it is worth it in this case though. I'll have a think.

Reply:
@ltrouton don't think too hard, we already did that for you 😂 
https://github.com/tinymce/tinymce/blob/ba6994f39b09ecd2c28cef68c3e8a85a4815e235/modules/katamari/src/main/ts/ephox/katamari/api/Adt.ts

Reply:
I am aware that exists in Katamari 😄. I know we largely moved away from using Adt in katamari in new code as it can be a bit un-wieldy to use and typed unions help a lot now

Reply:
I have done some refactoring to hopefully make everything cleaner